src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -171,6 +171,23 @@ protected function processValue($value, $isRoot = false)
                     continue;
                 }
 
+                if (0 !== \count($this->searchBindingByPropertyName($bindings, $parameter->name))) {
+                    $binding = $this->searchBindingByPropertyName($bindings, $parameter->name);
```

Comment:
the double search should be removed:
`if ($binding = $this->searchBindingByPropertyName($bindings, $parameter->name)) {`